### PR TITLE
[GH-692] Fix formatting inside code block in issue created notification

### LINKED ISSUE #692: Code in Pull Request code blocks is broken on Mattermost
For example, I just opened this Pull Request: https://github.com/indico/indico/issues/5905
The code block is just a standard triple-backtick-fenced block.

But in the Mattermost notification it became this:

```python
subcontribution_settings = EventSettingsProxy(&#39;subcontributions&#39;, {
    &#39;speakers_can_submit&#39;: False,  # Speakers can edit minutes and upload attachments
})
```

As you can see, the quotes got replaced with HTML entities.

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
Fixed the improper formatting of special characters inside the code block of issue-created channel notifications.

###### What to test?
- Special characters are properly displayed inside the code block present in the description of the issue-created notification.

#### Screenshots
Earlier:
image

Now:
image

#### Ticket Link
Fixes #692

### CODE REVIEW

**Review by hanzei — APPROVED** [review]

File: server/plugin/webhook.go [hunk 1]
```diff
@@ -459,7 +459,13 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
...
 	var policy = bluemonday.StrictPolicy()
 	policy.SkipElementsContent("details")
-	return strings.TrimSpace(policy.Sanitize(description))
+	// Replacing the unicodes with the respective special characters to have proper rendering in code block.
+	result := strings.ReplaceAll(policy.Sanitize(description), "&#39;", "'")
+	result = strings.ReplaceAll(result, "&#34;", "\"")
+	result = strings.ReplaceAll(result, "&amp;", "&")
+	result = strings.ReplaceAll(result, "&gt;", ">")
+	result = strings.ReplaceAll(result, "&lt;", "<")
+	return strings.TrimSpace(result)
```
Comment (mickmister) [inline server/plugin/webhook.go:468]: I wonder if there is a more catch-all holistic approach

**Review by mickmister — APPROVED** [review]

File: server/plugin/webhook.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/plugin/webhook.go:468]: @mickmister I explored other packages to sanitize the HTML but the one we are using right now was the best I could find. But this converts the special characters to their unicode value which are not formatted inside a code block, so we are replacing them again here. Please feel free to share your opinions if you have a better approach in mind.

Comment (mickmister) [reply server/plugin/webhook.go:468]: Hmm so it's the `policy.Sanitize` that's causing the issue. We really just want to use it to remove the `<details>` elements from the description, and not escape those characters.

As this comment https://github.com/microcosm-cc/bluemonday/issues/39#issuecomment-454358002 mentions on a related issue for the `bluemonday` package, we can call `html.UnescapeString` from the result to make those characters go back to what we want them to be

Comment (raghavaggarwal2308) [reply server/plugin/webhook.go]: @mickmister Updated

File: server/plugin/webhook.go [hunk 2]
```diff
@@ -459,12 +460,8 @@ func (p *Plugin) postPullRequestEvent(event *github.PullRequestEvent) {
...
 	var policy = bluemonday.StrictPolicy()
 	policy.SkipElementsContent("details")
-	// Replacing the unicodes with the respective special characters to have proper rendering in code block.
-	result := strings.ReplaceAll(policy.Sanitize(description), "&#39;", "'")
-	result = strings.ReplaceAll(result, "&#34;", "\"")
-	result = strings.ReplaceAll(result, "&amp;", "&")
-	result = strings.ReplaceAll(result, "&gt;", ">")
-	result = strings.ReplaceAll(result, "&lt;", "<")
+	result := policy.Sanitize(description)
+	html.UnescapeString(result)
```
Comment (mickmister) [inline server/plugin/webhook.go:464]: I think we need to reassign the value right?

```suggestion
	result = html.UnescapeString(result)
```

**Comment by mickmister** [comment]
Does the `bluemonday` policy remove things other than the `<details>` tags? I'm thinking it may remove things like `<script>` because the tool is meant to sanitize the HTML. We would not want it to remove anything but the `<details>` tags in our case.

**Comment by raghavaggarwal2308** [comment]
@mickmister Yes, it's removing the HTML tags in the string. For some tags, it's just removing the tags and for some, it's removing the tags along with the text inside.
issue_notification

**Comment by mickmister** [comment]
@raghavaggarwal2308 Since the main thing we want to sanitize is `<details>`, we could just only run the sanitize when there exists a `<details>` tag in the post body. It's not common to have those in the post body, but when it shows up it's usually spammy which is why we remove them

**Comment by raghavaggarwal2308** [comment]
> @raghavaggarwal2308 Since the main thing we want to sanitize is `<details>`, we could just only run the sanitize when there exists a `<details>` tag in the post body. It's not common to have those in the post body, but when it shows up it's usually spammy which is why we remove them

@mickmister If we do so and there are other tags along with the details tag, then they will be removed as well. Is that fine?

**Comment by mickmister** [comment]
@raghavaggarwal2308 Yes that's fine :+1:

**Comment by raghavaggarwal2308** [comment]
@mickmister Added logic to sanitize description conditionally.

**Review by mickmister — APPROVED** [review]

**Pull request merged**